lib/ecto/adapters/mysql/connection.ex

```diff
@@ -514,7 +514,7 @@ if Code.ensure_loaded?(Mariaex.Connection) do
 
     defp ecto_to_db(:string),     do: "varchar"
     defp ecto_to_db(:float),      do: "double"
-    defp ecto_to_db(:datetime),   do: "datetime"
+    defp ecto_to_db(:datetime),   do: "datetime(6)"
```

Comment:
One possible solution would be checking if the adapter supports microseconds. Something like: https://github.com/elixir-lang/ecto/commit/9860c362fa97a166b8868a97d9fcc87eb6bcca35, but I haven't checked if it's possible in this case.